### PR TITLE
Fix missing options when importing a connection object

### LINKED ISSUE #33: Missing `options` when importing a `connection` object
From terraform-provider-auth0 created by [atellezr](https://github.com/atellezr): alexkappa/terraform-provider-auth0#385  

### Description

When importing a `connection` object, the `options` are not imported.

I had a database connection defined in the Auth0 tenant, and I wanted to import it in Terraform. The database has custom scripts defined in the tenant, and what I wanted is to import everything related to that object into my local terraform configuration.

If I use the nodejs CLI (`auth0-deploy-cli`) I can get the connection configuration, including the `options` and, among others, the custom scripts.

And if I invoke directly the API `/api/v2/connections/con_AAAAAAAAAAAAAAAA` I can see the `options` with its attributes as well.

### Terraform Version

```
$ terraform -v
Terraform v0.14.4
+ provider registry.terraform.io/alexkappa/auth0 v0.20.0
```

(although there's a newer version, the v0.21.0, checking the code there are no significant differences related to the `options` object itself)

### Affected Resource(s)

* auth0_connection

### Terraform Configuration Files

At the moment of executing the import, the configuration that I have in Terraform related to the connection is only this one:

```hcl
resource "auth0_connection" "my_connection" {}
```

### Expected Behavior

I expect to import the `options` attribute with all its attributes (`mfa`, `customScripts`, ...), as part of the import process of the whole object.

### Actual Behavior

The `options` attribute is imported as an empty array. No child attributes are imported

### Steps to Reproduce

1. Prepare the Terraform configuration object (see _Terraform Configuration Files_ section)

2. Executed the import command:

```
$ terraform import auth0_connection.my_connection con_AAAAAAAAAAAAAAAA
auth0_connection.my_connection: Importing from ID "con_AAAAAAAAAAAAAAAA"...
auth0_connection.my_connection: Import prepared!
  Prepared auth0_connection for import
auth0_connection.lp_users: Refreshing state... [id=con_AAAAAAAAAAAAAAAA]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```

3. Verify the state of the imported object: it seems that the import went well, but when I have a look into the state (`terraform state show auth0_connection.my_connection`), I don't see the `options` attribute. In the Terraform state file I see the object as an empty array:

```
    {
      "mode": "managed",
      "type": "auth0_connection",
      "name": "my_connection",
      "provider": "provider[\"registry.terraform.io/alexkappa/auth0\"]",
      "instances": [
        {
          "schema_version": 2,
          "attributes": {
            "display_name": null,
            "enabled_clients": [
...
            ],
            "id": "con_AAAAAAAAAAAAAAAA",
            "is_domain_connection": false,
            "name": "MyConnection",
            "options": [],
...
```

### Debug Output

https://gist.github.com/atellezr/c8e336df73e3575ebf58e87f8ab4c676

### Panic Output

Not applicable

### Important Factoids

No, nothing remarkable

### References

N/A

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by sergiughf**
## Description

Fixes #33. 

## Checklist

**Note:** Checklist required to be completed before a PR is considered to be reviewable.

#### Auth0 Code of Conduct
- [x] I have read and agreed to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

#### Auth0 General Contribution Guidelines
- [x] I have read the [Auth0 General Contribution Guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)

#### Changes include test coverage?
- [x] Yes
- [ ] Not needed

#### Does the description provide the correct amount of context?
- [x] Yes, the description provides enough context for the reviewer to understand what these changes accomplish

#### Have you updated the documentation?
- [x] Yes, I've updated the appropriate docs
- [ ] Not needed

#### Is this code ready for production?
- [x] Yes, all code changes are intentional and no debugging calls are left over

### CODE REVIEW

File: auth0/structure_auth0_connection.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 )
 
 func flattenConnectionOptions(d ResourceData, options interface{}) []interface{} {
-	if options == nil || optionsBlockIsNotDefined(d) {
```
Comment (sergiughf) [inline auth0/structure_auth0_connection.go:11]: This is where Issue #33 gets fixed. 

After checking the terraform plugin SDK docs over [here](https://www.terraform.io/plugin/sdkv2/best-practices/detecting-drift), even though the options might be omitted from the terraform config we should always record it within the state in the READ function as the Auth0 API assigns a couple of default values to certain elements. 

This is also the reason why importing failed to get the options assigned within the state, although it was working fine after creating a new resource.

File: auth0/resource_auth0_connection_test.go [hunk 2]
```diff
@@ -1314,15 +1315,20 @@ func TestAccConnectionSAML(t *testing.T) {
 				Config: random.Template(testConnectionSAMLConfigCreate, rand),
 				Check: resource.ComposeTestCheckFunc(
 					random.TestCheckResourceAttr("auth0_connection.my_connection", "name", "Acceptance-Test-SAML-{{.random}}", rand),
+					random.TestCheckResourceAttr("auth0_connection.my_connection", "display_name", "Acceptance-Test-SAML-{{.random}}", rand),
 					resource.TestCheckResourceAttr("auth0_connection.my_connection", "strategy", "samlp"),
 					resource.TestCheckResourceAttr("auth0_connection.my_connection", "show_as_button", "false"),
-					random.TestCheckResourceAttr("auth0_connection.my_connection", "display_name", "Acceptance-Test-SAML-{{.random}}", rand),
+					resource.TestCheckResourceAttr("auth0_connection.my_connection", "options.#", "1"),
```
Comment (sergiughf) [inline auth0/resource_auth0_connection_test.go:1321]: We complement the test by making sure we also check the options are set as previously it was omitted.

File: auth0/resource_auth0_connection.go [hunk 3]
```diff
@@ -789,8 +790,6 @@ func readConnection(ctx context.Context, d *schema.ResourceData, m interface{})
 		return diag.FromErr(err)
 	}
 
-	d.SetId(auth0.StringValue(connection.ID))
```
Comment (sergiughf) [inline auth0/resource_auth0_connection.go:792]: From my understanding so far we should only set the ID of a resource when we create it, so no read method should set any ids as they are already set or if importing the `d.Id()` will be the id of the connection already.

**Review by sergiughf — COMMENTED** [review]
Leaving a few comments to aid with reviewing.

**Review by willvedd — APPROVED** [review]

**Pull request merged**